app/views/advanced_messenger/_html_head.html.erb

```diff
@@ -1,15 +1,218 @@
+<%# ============= Plugin styles  ============= %>
 <% content_for :header_tags do %>
     <%= stylesheet_link_tag 'advanced_messenger', :plugin => 'redmine_advanced_messenger' %>
 <% end %>  
+<%# ============= Global unread notifications indicator located on main title bar  ============= %>
 <%= javascript_tag do%>
-
-    function renderAllUnreadNotificationsIndicator(unreadNotificationsCount) {
+    function renderGlobalUnreadNotificationsIndicator(unreadNotificationsCount) {
         const color = unreadNotificationsCount > 0 ? "#ff4545" : "#2bff2b";
         $("#all-unread-notifications-indicator").html("<a href='/my/page'><%=t(:top_app_menu_unread_notifications)%>: <span style='color:" + color + "; font-weight:bold'>" + unreadNotificationsCount + "</span></a>");
     }
 
     jQuery( document ).ready(function() {
         $("div#loggedas").append("<span id='all-unread-notifications-indicator'><span>");
-        renderAllUnreadNotificationsIndicator(<%= getUnreadNotificationsForCurrentUserCount()%>);
+        renderGlobalUnreadNotificationsIndicator(<%= getUnreadNotificationsForCurrentUserCount()%>);
     });
+<% end %>
+<%# ============= Helper javascript render methods that are reused in many parts of the application ============= %>
+<%
+    current_controller = controller.controller_name
+    current_action = controller.action_name
+%>
+<% if current_controller == 'issues' && current_action == 'show' || current_controller == 'messages' && current_action == 'show' %>
+    <%= javascript_tag do %>
+        const RED = "red";
+        const GREEN = "#55CB85";
+
+        function renderExpandCollapseButton(expanded, currentNode, entityId, endpoint) {
+            const cssClass = expanded ? 'icon icon-toggle-minus' : 'icon icon-toggle-plus';
+            currentNode.find('.expand-collapse-icon-placeholder').html(
+                "<a class='" + cssClass + "' data-remote='true' rel='nofollow' data-method='post' href='/advanced_messenger/" + entityId + "/" + (expanded ? 2 : 1) + "/" + endpoint + "'></a>"
+            );
+        }
+
+        function renderExpandedCollapsed1(entityId, endpoint, currentNode, expanded, isInitialRender) {
```

Comment:
The `1` from the name of the function is a typo?

Reply:
Yes. I corrected it